Code/Framework/AzCore/AzCore/Settings/SettingsRegistryImpl.cpp

```diff
@@ -1463,58 +1296,64 @@ namespace AZ
             break;
         case Format::JsonMergePatch:
             mergeApproach = JsonMergeApproach::JsonMergePatch;
-            if (!jsonPatch.IsObject())
+            if (anchorKey.empty() && !jsonPatch.IsObject())
             {
-                AZStd::scoped_lock lock(LockForWriting());
-                pointer.Create(m_settings, m_settings.GetAllocator()).SetObject()
-                    .AddMember(StringRef("Error"), StringRef("Cannot merge registry file with a root which is not a JSON Object,"
-                        " an empty root key and a merge approach of JsonMergePatch. Otherwise the Settings Registry would be overridden."
-                        " See RFC 7386 for more information"), m_settings.GetAllocator())
-                    .AddMember(StringRef("Path"), Value(path, m_settings.GetAllocator()), m_settings.GetAllocator());
-
-                MergeSettingsResult result;
-                result.m_returnCode = MergeSettingsReturnCode::Failure;
-                result.m_operationMessages = AZStd::string::format(
-                    R"(Attempting to merge the settings registry file "%s" where the root element is a)"
-                    R"( non-JSON Object using the JSON MergePatch approach. The JSON MergePatch algorithm would therefore)"
+                mergeResult.Combine(MergeSettingsReturnCode::Failure);
+                mergeResult.m_operationMessages = AZStd::string::format(
+                    R"(Cannot merge settings registry JSON data to to root element field with a non-JSON Object)"
```

Comment:
spelling, I think this should be "due to" instead of "to to" ?

Reply:
Will fix.
```suggestion
                    R"(Cannot merge settings registry JSON data due to root element field with a non-JSON Object)"
```